cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/PrometheusMetricSampler.java

```diff
@@ -106,16 +107,14 @@ private void configurePrometheusAdapter(Map<String, ?> configs) {
         }
 
         try {
-            HttpHost host = HttpHost.create(endpoint);
-            if (host.getPort() < 0) {
-                throw new IllegalArgumentException();
-            }
+            String endpointWithScheme = endpoint.startsWith("http") ? endpoint : "http://" + endpoint;
+            URIBuilder uriBuilder = new URIBuilder(endpointWithScheme);
```

Comment:
For "http://example.com/path/to/query", its port is -1 in uriBuilder, but its default port is 80.

And IMO we don't need to verify if the port is >= 0, it is a common sense for software engineer.